src/content/developers/docs/oracles/index.md

```diff
@@ -1,314 +1,86 @@
 
-- Blockchain games and NFTs
-- Random assignment of duties and resources (e.g. randomly assigning judges to cases)
-- Choosing a representative sample for consensus mechanisms
+Oracles differ based on the source of data (one or multiple sources), trust models (centralized or decentralized) and system architecture (immediate-read, publish-subscribe, and request-response). We can also distinguish between oracles based on whether they retrieve external data for use by on-chain contracts (input oracles), send information from the blockchain to the real world (output orales), or perform computational tasks off-chain (computational oracles).
 
-Random numbers are difficult because blockchains are deterministic.
+## Why do smart contracts need oracles? {#why-do-smart-contracts-need-oracles}
 
-Working with Chainlink Oracles outside of data feeds follows the [request and receive cycle](https://docs.chain.link/docs/architecture-request-model) of working with Chainlink. They use the LINK token to send oracle providers oracle gas for returning responses. The LINK token is specifically designed to work with oracles and are based on the upgraded ERC-677 token, which is backwards compatible with [ERC-20](/developers/docs/standards/tokens/erc-20/).
-The following code, if deployed on the Kovan testnet will retrieve a cryptographically proven random number. To make the request, fund the contract with some testnet LINK token that you can get from the [Kovan LINK Faucet](https://kovan.chain.link/).
+Most developers see smart contracts as simply pieces of code running at specific addresses on the blockchain. However, a more [general view of smart contracts](/smart-contracts/) is that they are self-executing software programs capable of enforcing agreements between parties once specific conditions are met—which explains the term, “smart contracts.”
 
-```javascript
+But using smart contracts to enforce agreements between people isn't exactly straightforward, given that blockchains are designed to be “deterministic”.  A [deterministic system](https://en.wikipedia.org/wiki/Deterministic_algorithm) is one that always produces the same results given an initial state and a particular input—there is no randomness or variation in the process of computing outputs from inputs. 
 
-pragma solidity 0.6.6;
+To achieve deterministic execution, blockchains limit nodes to reaching consensus on simple binary (true/false) questions using *only* data stored on the blockchain itself. Examples of such questions include: “Did the account owner (identified by a public key) sign this transaction with the paired private key?”, “Does this account have enough funds to cover the transaction?”, “Is this transaction valid in the context of this smart contract?”, and so on. 
```

Comment:
To achieve deterministic execution, Ethereum limits nodes to reaching consensus on simple binary (true/false) questions using *only* data stored on the blockchain itself. Examples of such questions include: 

- Did the account owner (identified by a public key) sign this transaction with the paired private key?
- Does this account have enough funds to cover the transaction? 
- Is this transaction valid in the context of this smart contract?

Reply:
Added this now.